src/main/java/net/neoforged/neoforge/fluids/FluidStack.java

```diff
@@ -223,6 +260,13 @@ public FluidStack copy() {
         return new FluidStack(getFluid(), amount, tag);
     }
 
+    /**
+     * @return A copy of this FluidStack
+     */
+    public FluidStack copyWithAmount(int amount) {
+        return new FluidStack(getFluid(), amount, tag);
```

Comment:
The ItemStack equivalent `copyWithCount` returns the empty stack instance if the stack is currently empty. I followed the behavior of `FluidStack#copy()` which unlike `ItemStack#copy` doesn't have that check, but I am unsure if we potentially want to change it for `copyWithAmount`. Changing it for `copy` would potentially be a breaking change as then anyone who tries to set the amount on the returned stack would get an exception.

Reply:
IMO we should align the two in 1.20.5.

Reply:
We need to start a 1.20.5 TODO hackmd document so that we remember to do these changes and that modders know what to expect.

Reply:
I decided to add a TODO to the basic `copy` method for addressing it in 1.20.5, and I decided to just align the copy with amount method as it is new so might as well always match vanilla instead of changing in one version. I also when one step further than vanilla's check and also return the empty instance if the passed in amount is less than or zero.